src/lib/libc_preload/gen_syscalls_wrappers_c.py

```diff
@@ -0,0 +1,60 @@
+import urllib.request
+
+'''
+This script is meant to generate our interpose list containing all x86_64
+syscalls that are defined in the Linux kernel. The idea is that we can preload
+all syscalls even if Shadow doesn't yet support them all. Preloading is faster
+than using seccomp to interpose, so we should use preloading whenever possible.
+'''
+
+# Defines the latest syscalls.
+# See also https://github.com/torvalds/linux/tree/master/arch/x86/entry/syscalls
+syscall_tbl = 'https://raw.githubusercontent.com/torvalds/linux/master/arch/x86/entry/syscalls/syscall_64.tbl'
+
+with urllib.request.urlopen(syscall_tbl) as response:
+    data = response.read().decode("utf-8")
+
+syscalls = {}
+
+for line in data.splitlines():
+    parts = line.split()
+
+    # ignore comments and incomplete lines
+    if len(parts) < 4 or '#' in parts[0]:
+        continue
+
+    num, abi, name, entry = parts
+
+    # ignore the x32-specific abi, since shadow only supports x86_64
+    if 'x32' in abi:
+        continue
+
+    syscalls[str(name)] = [num, str(entry)]
+
+header = \
+'''// This file is generated with the 'generate_syscalls_c.py' script and in
+// general SHOULD NOT be edited manually.
```

Comment:
This file was renamed to `gen_syscalls_wrappers_c.py`.

Reply:
Good catch! Fixed.